assets/news-api/components/EditAPIToken.jsx

```diff
@@ -0,0 +1,279 @@
+                    },
+                    action: EDITOR_ACTIONS.CREATE,
+                });
+            });
+    }
+
+    onEnabledChange(event) {
+        this.setState({
+            token: {
+                ...this.state.token,
+                enabled: event.target.checked,
+            }
+        });
+    }
+
+    renderEditor() {
+        const existing = get(this.state, 'action') === EDITOR_ACTIONS.UPDATE;
+        const title = existing ?
+            gettext('Edit API Token') :
+            gettext('Create New API Token');
+
+        return (
+            <div className='tab-pane active company-api__token-edit' id='company-api-token'>
+                <form onSubmit={(event) => {event.preventDefault();}}>
+                    <div className="list-item__preview-form">
+                        <div className="card mt-3 d-block">
+                            <div className="card-header d-flex flex-row">{title}</div>
+                            <div className="card-body">
+                                <ExpiryDateInput
+                                    name="expiry"
+                                    label={gettext('Expires:')}
+                                    value={getDateInputDate(this.state.token.expiry)}
+                                    onChange={this.onExpiryChange}
+                                    error={get(this.state, 'errors.expiry')}
+                                />
+
+                                <CheckboxInput
+                                    name='is_enabled'
+                                    label={gettext('Enabled')}
+                                    value={this.state.token.enabled}
+                                    onChange={this.onEnabledChange}
+                                />
+                            </div>
+                            <div className="card-footer d-flex">
+                                <button
+                                    className="btn btn-outline-primary ml-auto"
+                                    onClick={this.saveToken}
+                                    disabled={Object.keys(this.state.errors).length > 0}
+                                >
+                                    {existing ? gettext('Save') : gettext('Generate Token')}
+                                </button>
+                                {existing && (
+                                    <button
+                                        className='btn btn-outline-secondary ml-3'
+                                        onClick={() => this.setState({action: EDITOR_ACTIONS.PREVIEW})}
+                                    >
+                                        {gettext('Cancel')}
+                                    </button>
+                                )}
+                            </div>
+                        </div>
+                    </div>
+                </form>
+            </div>
+        );
+    }
+
+    renderPreview() {
```

Comment:
Do we need the style of having two subcomponents for edit and preview? Can't we use the same form as with read-only attribute? Concerned as we continue to add components, we will have to create preview duplicates of it.